src/app/authentication/authentication.component.ts

```diff
@@ -39,12 +41,20 @@ export class AuthenticationComponent extends GraphExplorerComponent {
 
   // Https://docs.microsoft.com/en-us/azure/active-directory/active-directory-v2-protocols-implicit
   public login() {
-    this.authService.login();
-    }
+    localStorage.setItem('status', 'authenticating');
+    this.changeDetectorRef.detectChanges();
+    this.authService.login()
+      .then((accessToken) => {
+        if (accessToken) {
```

Comment:
When we fail to get a token (accessToken == false), we should set the **'status'** in localStorage to the default state (**anonymous**). This is the cause of the spinner that appears when we close the login poup.